xtable-core/src/main/java/org/apache/xtable/catalog/hms/IcebergHMSCatalogSyncHelper.java

```diff
@@ -0,0 +1,135 @@
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.api.hive_metastoreConstants;
+import org.apache.hadoop.security.UserGroupInformation;
+
+import org.apache.iceberg.BaseTable;
+import org.apache.iceberg.hadoop.HadoopTables;
+import org.apache.iceberg.mr.hive.HiveIcebergInputFormat;
+import org.apache.iceberg.mr.hive.HiveIcebergOutputFormat;
+import org.apache.iceberg.mr.hive.HiveIcebergSerDe;
+import org.apache.iceberg.mr.hive.HiveIcebergStorageHandler;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import org.apache.xtable.model.InternalTable;
+import org.apache.xtable.model.catalog.CatalogTableIdentifier;
+import org.apache.xtable.model.storage.TableFormat;
+
+class IcebergHMSCatalogSyncHelper {
+
+  private static final String ICEBERG_CATALOG_NAME_PROP = "iceberg.catalog";
+  private static final String ICEBERG_HADOOP_TABLE_NAME = "location_based_table";
+  private final HMSCatalogSyncClient syncClient;
+  private final HadoopTables hadoopTables;
+
+  IcebergHMSCatalogSyncHelper(HMSCatalogSyncClient syncClient) {
+    this.syncClient = syncClient;
+    this.hadoopTables = new HadoopTables(syncClient.getConfiguration());
+  }
+
+  @VisibleForTesting
+  IcebergHMSCatalogSyncHelper(HMSCatalogSyncClient syncClient, HadoopTables hadoopTables) {
+    this.syncClient = syncClient;
+    this.hadoopTables = hadoopTables;
+  }
+
+  Table getNewTable(InternalTable table, CatalogTableIdentifier tableIdentifier) {
+    try {
+      Table newTb = new Table();
+      newTb.setDbName(tableIdentifier.getDatabaseName());
+      newTb.setTableName(tableIdentifier.getTableName());
+      newTb.setOwner(UserGroupInformation.getCurrentUser().getShortUserName());
+      newTb.setCreateTime((int) ZonedDateTime.now().toEpochSecond());
+      newTb.setSd(getStorageDescriptor(table));
+      newTb.setTableType(TableType.EXTERNAL_TABLE.toString());
+      newTb.setParameters(getTableParameters(loadTableFromFs(table.getBasePath())));
+      return newTb;
+    } catch (IOException e) {
+      throw new RuntimeException(
+          "Failed to set owner for hms table: " + tableIdentifier.getId(), e);
+    }
+  }
+
+  Table getUpdatedTable(InternalTable table, Table catalogTable) {
+    BaseTable icebergTable = loadTableFromFs(table.getBasePath());
+    Table copyTb = new Table(catalogTable);
+    Map<String, String> parameters = copyTb.getParameters();
+    parameters.putAll(icebergTable.properties());
+    String currentMetadataLocation = parameters.get(METADATA_LOCATION_PROP);
+    parameters.put(PREVIOUS_METADATA_LOCATION_PROP, currentMetadataLocation);
+    parameters.put(METADATA_LOCATION_PROP, getMetadataFileLocation(icebergTable));
+    copyTb.setParameters(parameters);
+    copyTb
+        .getSd()
+        .setCols(
+            syncClient.getSchemaExtractor().toColumns(TableFormat.ICEBERG, table.getReadSchema()));
```

Comment:
If you pass this as an argument from caller you can avoid needing syncClient as a field in this class and a separate class for iceberg. If you look at closely these methods will have similar implementation for delta and hudi except different parameters, think about it and let's discuss.

Reply:
the reason for passing syncClient itself is to access configuration and schemaExtractor from a single field instead of passing them separately. In future, if we require any other objects, we can initialize them in client and use it instead of updating method definitions